rules/framework.bzl

```diff
@@ -699,70 +726,130 @@ def _bundle_dynamic_framework(ctx, is_extension_safe, avoid_deps):
         attr = ctx.attr,
         res_attrs = ["infoplists"],
     )
-    link_result = linking_support.register_binary_linking_action(
-        ctx,
-        avoid_deps = avoid_deps,
-        entitlements = None,
-        extra_linkopts = extra_linkopts,
-        platform_prerequisites = platform_prerequisites,
-        stamp = ctx.attr.stamp,
-    )
+    if apple_api_version == "3.0":
+        link_result = linking_support.register_binary_linking_action(
+            ctx,
+            avoid_deps = avoid_deps,
+            entitlements = None,
+            exported_symbols_lists = ctx.files.exported_symbols_lists,
+            extra_linkopts = extra_linkopts,
+            platform_prerequisites = platform_prerequisites,
+            stamp = ctx.attr.stamp,
+        )
+    else:
+        link_result = linking_support.register_binary_linking_action(
+            ctx,
+            avoid_deps = avoid_deps,
+            entitlements = None,
+            extra_linkopts = extra_linkopts,
+            platform_prerequisites = platform_prerequisites,
+            stamp = ctx.attr.stamp,
+        )
+
     binary_artifact = link_result.binary
     debug_outputs = linking_support.debug_outputs_by_architecture(link_result.outputs)
 
-    archive_for_embedding = outputs.archive_for_embedding(
-        actions = actions,
-        bundle_name = bundle_name,
-        bundle_extension = bundle_extension,
-        executable_name = bundle_name,
-        label_name = label.name,
-        rule_descriptor = rule_descriptor,
-        platform_prerequisites = platform_prerequisites,
-        predeclared_outputs = predeclared_outputs,
-    )
-
-    dep_frameworks = ctx.attr.frameworks
-
-    # TODO(jmarino) - consider how to better handle frameworks of frameworks
-    processor_partials = []
-
-    processor_partials.append(
-        partials.apple_bundle_info_partial(
+    if apple_api_version == "3.0":
+        archive_for_embedding = outputs.archive_for_embedding(
             actions = actions,
-            bundle_extension = bundle_extension,
-            bundle_id = bundle_id,
             bundle_name = bundle_name,
-            executable_name = bundle_name,
+            bundle_extension = bundle_extension,
```

Comment:
Is `executable_name` not needed here or was it removed?

Reply:
Yeah it was removed